src/Microsoft.Health.Api/Features/HealthChecks/HealthCheckTimeoutPostConfigure.cs

```diff
@@ -0,0 +1,30 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Health.Api.Features.HealthChecks
+{
+    internal sealed class HealthCheckTimeoutPostConfigure : IPostConfigureOptions<HealthCheckServiceOptions>
+    {
+        private readonly TimeSpan _timeout;
+
+        public HealthCheckTimeoutPostConfigure(TimeSpan timeout)
+            => _timeout = EnsureArg.IsGte(timeout, TimeSpan.Zero, nameof(timeout));
+
+        public void PostConfigure(string name, HealthCheckServiceOptions options)
```

Comment:
Is there any order to the execution of the two post configure options?

Reply:
Or I guess more so, is there any guarantee that they'll be registered in the correct order?

Reply:
I am unsure if there is a way to guarantee order... I'll check! At this point they are presumably based on the order in which they are added to the service container

Reply:
Removed timeout from the cache specifically